src/main/java/io/confluent/connect/jdbc/dialect/PostgreSqlDatabaseDialect.java

```diff
@@ -342,4 +342,39 @@ protected void formatColumnValue(
     }
   }
 
+  @Override
+  protected int decimalScale(ColumnDefinition defn) {
+    if (defn.scale() == NUMERIC_TYPE_SCALE_UNSET) {
+      return NUMERIC_TYPE_SCALE_HIGH;
+    }
+
+    // Postgres requires DECIMAL/NUMERIC columns to have a precision greater than zero
+    // If the precision appears to be zero, it's because the user didn't define a fixed precision
+    // for the column.
+    if (defn.precision() == 0) {
+      // In that case, a scale of zero indicates that there also isn't a fixed scale defined for
+      // the column. Instead of treating that column as if its scale is actually zero (which can
+      // cause issues since it may contain values that aren't possible with a scale of zero, like
+      // 12.12), we fall back on NUMERIC_TYPE_SCALE_HIGH to try to avoid loss of precision
+      if (defn.scale() == 0) {
+        log.debug(
+            "Column {} does not appear to have a fixed scale defined; defaulting to {}",
+            defn.id(),
+            NUMERIC_TYPE_SCALE_HIGH
+        );
+        return NUMERIC_TYPE_SCALE_HIGH;
```

Comment:
I suppose it could, but this is a pattern we already follow in other cases: https://github.com/confluentinc/kafka-connect-jdbc/blob/348e49f4a93261dd7b1365f469af426fa9ee8c6c/src/main/java/io/confluent/connect/jdbc/dialect/GenericDatabaseDialect.java#L1489-L1491

IIRC think that particular logic was put in place for Oracle Database, which has a default precision of 38 for `NUMERIC` types.

We could consider refining the logic for if/how we provide the precision of logical Connect `Decimal` types in cases like this, but it may be premature to block this fix on that refinement given that it's only following precedent already set by the connector in similar cases, and even without improving how we provide a precision for logical decimal types, this change should still be beneficial for users. What are your thoughts?

Reply:
I agree with @C0urante that this works and is better than what we already have as we set the scale to `NUMERIC_TYPE_SCALE_HIGH` only when precision is 0, and scale being greater than precision should not be the case when the column has a fixed/ defined precision and scale.

Reply:
> refining the logic for if/how we provide the precision of logical Connect Decimal types in cases like this

That sounds fine. We have fixed this already in the Oracle CDC connector and would eventually port the logic over to JDBC. This would just be one more location to apply the fix to when we do the port over.